manticore/utils/config.py

```diff
@@ -0,0 +1,208 @@
+"""
+This file implements a configuration system.
+
+The config values and constant are gathered from three sources:
+
+    1. default values provided at time of definition
+    2. ini files (i.e. ./manticore.ini)
+    3. command line arguments
+
+in that order of priority.
+"""
+
+import ast
+import configparser
+import io
+import logging
+import os
+
+from itertools import product
+
+
+_groups = {}
+
+
+logger = logging.getLogger(__name__)
+
+
+class ConfigError(RuntimeError):
+    pass
+
+
+class _var:
+    def __init__(self, default=None, description: str=None, defined: bool=True):
+        self.description = description
+        self.value = default
+        self.default = default
+        self.defined = defined
+
+    @property
+    def was_set(self) -> bool:
+        return self.value is not self.default
+
+
+class _group:
```

Comment:
I am not 100% sure but it seems we could inherit from a dict and remove some of the code